### PR TITLE
skip empty event

### PR DESCRIPTION
**Pull request by sirenkovladd**
Fixes: https://github.com/tinylibs/tinybench/issues/54

### CODE REVIEW

**Comment by sirenkovladd** [comment]
@Uzlopak @kibertoad @Aslemammad

File: test/index.test.ts [hunk 1]
```diff
@@ -366,3 +366,12 @@ test('throw error in beforeAll, afterAll, beforeEach, afterEach', async () => {
...
   expect(bench.getTask('AA test')!.result!.error).toBe(AAerror);
 });
+
+test('remove non-existent task', () => {
+  const bench = new Bench();
+  bench.addEventListener('remove', () => {
+    throw new Error('should not be called');
+  });
+
+  bench.remove('non-existent');
```
Comment (Uzlopak) [inline test/index.test.ts:376]: check how vitest tests for non throwing.

File: test/index.test.ts [hunk 2]
```diff
@@ -366,3 +366,12 @@ test('throw error in beforeAll, afterAll, beforeEach, afterEach', async () => {
   expect(bench.getTask('AE test')!.result!.error).toBe(AEerror);
   expect(bench.getTask('AA test')!.result!.error).toBe(AAerror);
 });
+
+test('remove non-existent task', () => {
```
Comment (Uzlopak) [inline test/index.test.ts]: try to make the description more descriptive.

```suggestion
test('removing non-existing task should not throw', () => {
```

**Comment by Uzlopak** [comment]
The question, which has to be discussed before hand is, if it makes sense to ignore the non existing task, or to throw a custom error.

File: test/index.test.ts (hunk 1 above)
Comment (sirenkovladd) [reply test/index.test.ts:376]: No errors, just an empty event received in the listener

**Comment by sirenkovladd** [comment]
I don't have much respect for throwing errors
I suggest returning boolean instead of `this`

**Comment by Uzlopak** [comment]
with returning `this` we allow chaining/ train wreck pattern.

**Comment by sirenkovladd** [comment]
are you sure adding `remove` to the chain is a good idea?

File: test/index.test.ts (hunk 1 above)
Comment (sheremet-va) [reply test/index.test.ts:376]: There is `expect.unreachable()` for situations like this

Comment (sirenkovladd) [reply test/index.test.ts:376]: Approval required to update version for "vitest"

**Review by Uzlopak — APPROVED** [review]
LGTM

File: test/index.test.ts (hunk 1 above)
Comment (Uzlopak) [reply test/index.test.ts:376]: Update to latest vitest if necessary

**Pull request merged**